File: python/blur.py
import math
import random
import time

WIDTH = 500
HEIGHT = 500
SIZE = WIDTH * HEIGHT
RADIUS = 32

band = []

for i in range(0, SIZE):
    band.append(math.floor(random.randint(0, 255)))

print(band[0: 100], "...", SIZE - 100, "more items")

newBand = []

start = time.time()

i = 0

for h in range(0, HEIGHT):
    for w in range(0, WIDTH):
        cellCount = 0
        sumOfNumbers = 0

        for m in range(-RADIUS, RADIUS + 1):
            for n in range(-RADIUS, RADIUS + 1):
                if not (m + h >= HEIGHT or m + h <= -1 or n + w >= WIDTH or n + w <= -1):
                    cellCount += 1
                    sumOfNumbers += band[i + (WIDTH * m + n)]

        newBand.append(round(sumOfNumbers / cellCount))

elapsed = time.time() - start

print("Time taken:", elapsed, "seconds")
print(newBand[0: 100], "...", SIZE - 100, "more items")

File: js/blur.js
const width = 500,
  height = 500,
  size = width * height,
  radius = 32;

let band = [];

for (let i = 0; i < size; i++) {
  band.push(Math.floor(Math.random() * 255));
}

console.log("Original:", band);

function main() {
  let newBand = [];

  let start = Date.now();

  for (let i = 0, h = 0; h < height; h++) {
    for (let w = 0; w < width; w += 1, i += 1) {
      let cellCount = 0,
        sum = 0;

      for (let m = -radius; m <= radius; m++) {
        for (let n = -radius; n <= radius; n++) {
          if (m + h < height && m + h > -1 && n + w < width && n + w > -1) {
            cellCount++;
            sum += band[i + (width * m + n)];
          }
        }
      }

      newBand.push(Math.round(sum / cellCount));
    }
  }

  let elapsed = (Date.now() - start) / 1000;

  console.log("Time taken:", elapsed, "seconds");
  console.log("Output:", newBand);
}

for (let i = 0; i < 10; i++) {
  main();
}
